Sming/SmingCore/Network/Ssl/SslValidator.h

```diff
@@ -16,20 +16,40 @@
 #ifndef _SMING_CORE_SSLVALIDATOR_H_
 #define _SMING_CORE_SSLVALIDATOR_H_
 
-#ifdef ENABLE_SSL
-
 #include "ssl/ssl.h"
 #include "ssl/tls1.h"
 
 #include <functional>
+#include "WVector.h"
 
 typedef std::function<bool(SSL* ssl, void* data)> SslValidatorCallback;
 
 bool sslValidateCertificateSha1(SSL* ssl, void* data);
 
 bool sslValidatePublicKeySha256(SSL* ssl, void* data);
 
-#endif /* ENABLE_SSL */
+struct SSLValidator {
+	SslValidatorCallback callback;
+	void* data; ///< Fingerprint
+};
+
+class SSLValidatorList : private Vector<SSLValidator>
```

Comment:
Any advantage to using a separate list for validation? Could this be wrapped into `SslFingerprints`?